### PR TITLE
AI goals refactoring

### PR DESCRIPTION
**Pull request by nullkiller**
Wanna split Goals.cpp onto small files

### CODE REVIEW

**Comment by dydzio0614** [comment]
While you work on it, you may want on Goals namespace if it is really needed and dealwiththe mess where some goal class methods are in goal namespace andsome are not,  sofor example in visual studio you need to check members of Goals::VisitObj to find constructor and VisitObj without goals to get rest of methods. Some goals may be worth renaming to reflect their purpose clearly - for example GatherArmy and GatherTroops.

File: AI/VCAI/Goals/AbstractGoal.h [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+			value = 0;
+			aid = -1;
+			objid = -1;
+			resID = -1;
+			tile = int3(-1, -1, -1);
+			town = nullptr;
+		}
+
+		OSETTER(bool, isElementar)
+			OSETTER(bool, isAbstract)
```
Comment (alexvins) [inline AI/VCAI/Goals/AbstractGoal.h]: wrong indentation

File: AI/VCAI/Goals/Build.cpp [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+using namespace Goals;
+
+TGoalVec Goals::Build::getAllPossibleSubgoals()
+{
+	TGoalVec ret;
+
+	for(const CGTownInstance * t : cb->getTownsInfo())
+	{
+		//start fresh with every town
+		ai->ah->getBuildingOptions(t);
```
Comment (alexvins) [inline AI/VCAI/Goals/Build.cpp]: Is return value ignored intentionally?

File: AI/VCAI/Goals/CollectRes.h [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+		CollectRes()
+			: CGoal(Goals::COLLECT_RES)
+		{
+		}
+		CollectRes(int rid, int val)
+			: CGoal(Goals::COLLECT_RES)
+		{
+			resID = rid;
+			value = val;
+			priority = 2;
```
Comment (alexvins) [inline AI/VCAI/Goals/CollectRes.h]: It should be on common header with all that priority constants.

File: AI/VCAI/Goals/Build.cpp (hunk 2 above)
Comment (nullkiller) [reply AI/VCAI/Goals/Build.cpp]: I have not changed any logic. Just mechanical split of one big file.

File: AI/VCAI/Goals/CollectRes.h (hunk 3 above)
Comment (nullkiller) [reply AI/VCAI/Goals/CollectRes.h]: Can put them in abstractgoal.h or in fuzzy helper (it is actually responsible for priority). technically they should not be hardcoded

File: AI/VCAI/Goals/Build.cpp (hunk 2 above)
Comment (nullkiller) [reply AI/VCAI/Goals/Build.cpp]: I was right. It fetches to internal state. It does not look like Get*

**Comment by nullkiller** [comment]
I just want to split the big file. I do not plan any deep refactoring.

File: AI/VCAI/Goals/CollectRes.cpp [hunk 4]
```diff
@@ -0,0 +1,208 @@
+/*
+ * Goals.cpp, part of VCMI engine
```
Comment (ArseniyShestakov) [inline AI/VCAI/Goals/CollectRes.cpp]: Please fix all filenames in headers so we don't have to do it after.

Comment (nullkiller) [reply AI/VCAI/Goals/CollectRes.cpp]: Ok, will do

**Comment by ArseniyShestakov** [comment]
Tested and this seems to be working fine for me though it's expected. Merged.

**Pull request merged**